### PR TITLE
JUG-1 Add push container image workflow

### PR DESCRIPTION
**Pull request by datalek**
#### List of Changes

- Introduce a shared action
- Add push-image github workflow
- Refactor `main` github workflow

#### Motivation and Context

Every time a tag is added, the container image needs to be pushed.

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: .github/workflows/push-image.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+      - uses: ./.github/actions/build-image
+        with:
+          image_tag: ${{ env.image_tag }}
+
+  deliver:
+    name: Build and push container image
+    needs: build-container
+    # This job is trigger only on new tags
+    if: startsWith(github.ref, 'refs/tags/')
```
Comment (kin0992) [inline .github/workflows/push-image.yaml:28]:
```suggestion
```
This should not be necessary, right? This workflow is triggered only with a push of a tag, so the condition is redundant

File: .github/actions/build-image/action.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+runs:
+  using: "composite"
+  steps:
+    - uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab
+
+    - name: Build container image
+      uses: docker/build-push-action@2eb1c1961a95fc15694676618e422e8ba1d63825
+      with:
+        tags: ${{ env.image_tag }}
```
Comment (kin0992) [inline .github/actions/build-image/action.yaml:17]: Just a curiosity: is there any difference to access the `image_tag` property using `env.image_tag` or `inputs.image_tag`?

File: .github/workflows/push-image.yaml (hunk 1 above)
Comment (datalek) [reply .github/workflows/push-image.yaml:28]: Yes right!

File: .github/actions/build-image/action.yaml (hunk 2 above)
Comment (datalek) [reply .github/actions/build-image/action.yaml:17]: Honestly I do not know. It should be `inputs.image_tag`. I'm going to change it.

Comment (kin0992) [inline .github/actions/build-image/action.yaml:17]:
```suggestion
        tags: ${{ inputs.image_tag }}
```
I guess this should change as well

**Review by kin0992 — CHANGES_REQUESTED** [review]

**Review by kin0992 — APPROVED** [review]
👽

**Review by kin0992 — APPROVED** [review]
🏞️

**Pull request merged**